es5-persistence/src/main/java/com/netflix/conductor/dao/es5/index/ElasticSearchRestDAOV5.java

```diff
@@ -653,27 +661,40 @@ private void indexObject(final String index, final String docType, final String
         IndexRequest request = new IndexRequest(index, docType, docId);
         request.source(docBytes, XContentType.JSON);
 
-        indexWithRetry(request, "Indexing " + docType + ": " + docId);
+        if(bulkRequests.get(docType) == null) {
+            bulkRequests.put(docType, new BulkRequest());
+        }
+
+        bulkRequests.get(docType).add(request);
+        if (bulkRequests.get(docType).numberOfActions() >= this.indexBatchSize) {
+            indexWithRetry(bulkRequests.get(docType), "Indexing " + docType + ": " + docId);
+            bulkRequests.put(docType, new BulkRequest());
+        }
     }
 
     /**
      * Performs an index operation with a retry.
      * @param request The index request that we want to perform.
      * @param operationDescription The type of operation that we are performing.
      */
-    private void indexWithRetry(final IndexRequest request, final String operationDescription) {
+    private synchronized void indexWithRetry(final BulkRequest request, final String operationDescription) {
```

Comment:
@apanicker-nflx Sometime I see ConcurrentModificationException with List based approach. So making this method synchronized